GVFS/GVFS.Common/Database/PlaceholderTable.cs

```diff
@@ -158,6 +155,37 @@ public void AddPossibleTombstoneFolder(string path)
             this.Insert(new PlaceholderData() { Path = path, PathType = PlaceholderData.PlaceholderType.PossibleTombstoneFolder });
         }
 
+        public List<IPlaceholderData> RemoveStartingWith(string path)
+        {
+            // Normalize the path to match what will be in the database
+            path = NormalizePath(path);
+
+            try
+            {
+                using (IDbConnection connection = this.connectionPool.GetConnection())
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    List<IPlaceholderData> removedPlaceholders = new List<IPlaceholderData>();
+                    command.CommandText = "SELECT path, pathType, sha FROM Placeholder WHERE path LIKE @path";
+                    command.AddParameter("@path", DbType.String, $"{path}%");

```

Comment:
As best I can tell the normalized path for folders will not have a trailing directory separator (the final call on the path is `Trim(Path.DirectorySeparatorChar)`).

I think this means that if `RemoveStartingWith` were called for the folder `Foo`, this query would also return placeholders like `FooBar\baz.txt`

If there's not one already, a functional test for this scenario would be good.